Inner exception: Member 'ListViewItemCount' was not found.

<!--
If you are new to the project get yourself familiar with https://www.mono-project.com/community/bugs/make-a-good-bug-report/ before filling the issue
-->

## Steps to Reproduce

1. Clone the following MWE: https://github.com/LeonardKoenig/listview_mono_mwe
2. Run `msbuild`. Alternatively directly run
   ```
   $ resgen /useSourcePath /compile Form1.resx,obj/Debug/ListViewTest.Form1.resources
   ```
   in the `ListViewTest` directory.


<!--
You may drag & drop the attachment (repro code/solution, screenshot, etc.) onto the issue.
-->

### Current Behavior

<!--
What is the current behavior?
-->

The build fails with:

```
CoreResGen:
  /usr/lib/mono/4.5/resgen.exe /useSourcePath /r:/usr/lib/mono/4.6.2-api/Microsoft.CSharp.dll /r:/usr/lib/mono/4.6.2-api/mscorlib.dll /r:/usr/lib/mono/4.6.2-api/System.Core.dll /r:/usr/lib/mono/4.6.2-api/System.Data.DataSetExtensions.dll /r:/usr/lib/mono/4.6.2-api/System.Data.dll /r:/usr/lib/mono/4.6.2-api/System.Deployment.dll /r:/usr/lib/mono/4.6.2-api/System.dll /r:/usr/lib/mono/4.6.2-api/System.Drawing.dll /r:/usr/lib/mono/4.6.2-api/System.Net.Http.dll /r:/usr/lib/mono/4.6.2-api/System.Windows.Forms.dll /r:/usr/lib/mono/4.6.2-api/System.Xml.dll /r:/usr/lib/mono/4.6.2-api/System.Xml.Linq.dll /compile Form1.resx,obj/Debug/ListViewTest.Form1.resources Properties/Resources.resx,obj/Debug/ListViewTest.Properties.Resources.resources
  Processing resource file "Form1.resx" into "obj/Debug/ListViewTest.Form1.resources".
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error MSB3103: Invalid Resx file. Exception has been thrown by the target of an invocation. [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error : TargetInvocationException: Exception has been thrown by the target of an invocation. [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error :   at System.Resources.ResXResourceReader.LoadData () [0x0010c] in <f73fa304d089422f9e23b25394111843>:0  [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error :   at System.Resources.ResXResourceReader.GetEnumerator () [0x00008] in <f73fa304d089422f9e23b25394111843>:0  [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error :   at Microsoft.Build.Tasks.ProcessResourceFiles.ReadResources (Microsoft.Build.Tasks.ProcessResourceFiles+ReaderInfo readerInfo, System.Resources.IResourceReader reader, System.String fileName) [0x00002] in <f6dbc5d4497740418a0ffd4a9d5d751e>:0  [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error :   at Microsoft.Build.Tasks.ProcessResourceFiles.ReadResources (System.String filename, System.Boolean shouldUseSourcePath, System.String outFileOrDir) [0x00085] in <f6dbc5d4497740418a0ffd4a9d5d751e>:0  [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error :   at Microsoft.Build.Tasks.ProcessResourceFiles.ProcessFile (System.String inFile, System.String outFileOrDir) [0x00074] in <f6dbc5d4497740418a0ffd4a9d5d751e>:0  [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error : SerializationException: Member 'ListViewItemCount' was not found. [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error :   at System.Runtime.Serialization.SerializationInfo.GetElement (System.String name, System.Type& foundType) [0x00020] in <4f52a2659e1844fea7ab38854dcbcf68>:0  [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error :   at System.Runtime.Serialization.SerializationInfo.GetInt32 (System.String name) [0x00000] in <4f52a2659e1844fea7ab38854dcbcf68>:0  [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error :   at System.Windows.Forms.ListViewGroup..ctor (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context) [0x0006a] in <f73fa304d089422f9e23b25394111843>:0  [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error :   at (wrapper managed-to-native) System.Reflection.RuntimeConstructorInfo.InternalInvoke(System.Reflection.RuntimeConstructorInfo,object,object[],System.Exception&) [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error :   at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x00005] in <4f52a2659e1844fea7ab38854dcbcf68>:0  [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
/tmp/listview_mono_mwe/ListViewTest/Form1.resx : error :  [/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj]
  Processing resource file "Properties/Resources.resx" into "obj/Debug/ListViewTest.Properties.Resources.resources".
Done Building Project "/tmp/listview_mono_mwe/ListViewTest/ListViewTest.csproj" (default targets) -- FAILED.

Build FAILED.
```

or, when just running `resgen` directly, with:

```
Error: Exception has been thrown by the target of an invocation.
Inner exception: Member 'ListViewItemCount' was not found.
```

### Expected Behavior

<!--
Please describe the behavior you are expecting
-->

The resource file(s) should be generated.

## On which platforms did you notice this

 - [ ] macOS (couldn't test)
 - [x] Linux
 - [ ] Windows (couldn't test)
 - [ ] Windows with MS .NET

**Version Used**:

<!--
You can use `mono --version` or About dialog to obtain this information.
-->

```
$ mono --version
Mono JIT compiler version 6.0.0 (makepkg/6256b82d62f Fri 20 Sep 2019 11:18:59 PM -03)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug
        Interpreter:   yes
        LLVM:          supported, not enabled.
        Suspend:       hybrid
        GC:            sgen (concurrent by default)
```

<!--
You can join us at https://gitter.im/mono/mono to discuss your reported issue
-->

----

The issue seems to be with de-serializing the base64-encoded `System.Windows.Forms.ListViewGroup` property.

The issue was also reported on the old bugtracker here: https://xamarin.github.io/bugzilla-archives/40/40237/bug.html

--- a/mcs/class/System.Windows.Forms/System.Windows.Forms/ListViewGroup.cs
+++ b/mcs/class/System.Windows.Forms/System.Windows.Forms/ListViewGroup.cs
@@ -86,16 +86,16 @@ namespace System.Windows.Forms
 
 			int count = 0;
 			try {
-        			count = info.GetInt32("ItemsCount");
+				count = info.GetInt32("ItemsCount");
 			} catch (SerializationException e) {
 				// Mono backwards compat
 				try {
-        				count = info.GetInt32("ListViewItemCount");
-				} catch (SerializationException e) {}
+ 					count = info.GetInt32("ListViewItemCount");
+				} catch (SerializationException e2) {}
 			}
 
 			if (items == null) {
-        			items = new ListView.ListViewItemCollection(list_view_owner);
+				items = new ListView.ListViewItemCollection(list_view_owner);
 			}
 
 			for (int i = 0; i < count; i++)

--- a/mcs/class/System.Windows.Forms/System.Windows.Forms/ListViewGroup.cs
+++ b/mcs/class/System.Windows.Forms/System.Windows.Forms/ListViewGroup.cs
@@ -84,15 +84,23 @@ namespace System.Windows.Forms
 			header_alignment = (HorizontalAlignment)info.GetInt32("HeaderAlignment");
 			tag = info.GetValue("Tag", typeof(object));
 
-			int count = info.GetInt32("ListViewItemCount");
-			if (count > 0) {
-				if(items == null)
-				items = new ListView.ListViewItemCollection(list_view_owner);
+			int count = 0;
+			try {
+        			count = info.GetInt32("ItemsCount");
+			} catch (SerializationException e) {
+				// Mono backwards compat
+				try {
+        				count = info.GetInt32("ListViewItemCount");
+				} catch (SerializationException e) {}
+			}
 
-				for (int i = 0; i < count; i++)
-				{
-					items.Add((ListViewItem)info.GetValue(string.Format("ListViewItem_{0}", i), typeof(ListViewItem)));
-				}
+			if (items == null) {
+        			items = new ListView.ListViewItemCollection(list_view_owner);
+			}
+
+			for (int i = 0; i < count; i++)
+			{
+				items.Add((ListViewItem)info.GetValue(string.Format("ListViewItem_{0}", i), typeof(ListViewItem)));
 			}
 		}
 

